tests/backup/backup_psa_test.go

```diff
@@ -548,3 +551,597 @@ var _ = Describe("{EnableNsAndClusterLevelPSAWithBackupAndRestore}", Label(TestC
+			namespaces = append(namespaces, restrictedNamespace)
+			Inst().AppList = psaApp
+
+			currentScheduledAppContexts := make([]*scheduler.Context, 0)
+			for i := 0; i < numberOfBackups; i++ {
+				taskName := fmt.Sprintf("%s-%v-%d", TaskNamePrefix, RandomString(3), i)
+				scheduledAppContextsTemp := ScheduleApplicationsOnNamespace(restrictedNamespace, taskName)
+				currentScheduledAppContexts = append(currentScheduledAppContexts, scheduledAppContextsTemp...)
+			}
+			log.InfoD("Validating the deployed applications")
+			ValidateApplications(currentScheduledAppContexts)
+			scheduledAppContexts = append(scheduledAppContexts, currentScheduledAppContexts...)
+
+			err = CreateBackupWithValidation(ctx, multiNSBackupNameWithVolumeAndResourceWithRestrictedLabel, SourceClusterName, backupLocationName, backupLocationUID, append(restrictedScheduledAppContexts, currentScheduledAppContexts...), nil, BackupOrgID, srcClusterUid, preRuleName, preRuleUid, postRuleName, postRuleUid)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of backup of multiple namespaces with restricted PSA label with volume and resources [%s]", multiNSBackupNameWithVolumeAndResourceWithRestrictedLabel))
+			backupList = append(backupList, multiNSBackupNameWithVolumeAndResourceWithRestrictedLabel)
+
+			// Switching cluster context to destination cluster
+			log.InfoD("Switching cluster context to destination cluster")
+			err = SetDestinationKubeConfig()
+			log.FailOnError(err, "Failed to set destination kubeconfig")
+
+			// Create namespaces with baseline PSA type to take restore in that namespace
+			baselineNamespace1 := fmt.Sprintf("single-ns-multi-app-%v-baseline-label", RandomString(3))
+			baselineNamespace2 := fmt.Sprintf("single-ns-multi-app-%v-baseline-label", RandomString(3))
+
+			for _, baselineNamespace := range []string{baselineNamespace1, baselineNamespace2} {
+				err = CreateNamespaceAndAssignLabels(baselineNamespace, BaselinePSALabel)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating namespace %v and assigning baseline labels", baselineNamespace))
+			}
+
+			log.InfoD("Restoring backup of multiple namespaces with restricted PSA label to baseline namespaces with namespace & storage class mapping to namespace with baseline PSA set")
+			params["repl"] = "2"
+			for _, sc := range sourceScNameList {
+				scName := fmt.Sprintf("replica-sc-%v", RandomString(3))
+				v1obj := metaV1.ObjectMeta{
+					Name: scName,
+				}
+				reclaimPolicyDelete := v1.PersistentVolumeReclaimDelete
+				bindMode := storageApi.VolumeBindingImmediate
+				scObj := storageApi.StorageClass{
+					ObjectMeta:        v1obj,
+					Provisioner:       k8s.CsiProvisioner,
+					Parameters:        params,
+					ReclaimPolicy:     &reclaimPolicyDelete,
+					VolumeBindingMode: &bindMode,
+				}
+				log.InfoD("Create new storage class on destination cluster for storage class mapping for restore")
```

Comment:
@vpinisetti-px  should we move this to create restore function as discussed to avoid duplicate code in many testcases

Reply:
ok we can ignore this comment ,since here it is different use case